src/main.cpp

```diff
@@ -428,6 +428,10 @@ int main(int argc, char *argv[])
 
     // Now we do the actual game.
 
+#ifdef ZMQ
```

Comment:
It would be better to move those into the publish header and implement an compatible class that will be used when `ZMQ` is not defined, something like:
```C++
#if ZMQ
// implementation with ZMQ
#else
class publish {
public:
    // compatible but empty interface.
    publish(const char *) { }
    void send( const std::string &, const std::string & ) { }
};
#endif
```
Or move the define check inside the class:
```C++
class publish {
#if ZMQ
...
#else
    publish(const char *) { }
    void send( const std::string &, const std::string & ) { }
#endif
};
```
That way you can use the class without having to check for the `ZMQ` define.